app/views/site/_search_options.html.erb

```diff
@@ -1,173 +1,188 @@
-          <div id="spinner_target"></div>
+      </div>
+      <hr />
+      <% if action_name =~ /(gene_set|heatmap)/ && action_name !~ /precomputed/ %>
+          <div class="row">
+            <div class="col-sm-12">
+              <%= label_tag :gene, 'Load single gene-level expression' %><br />
+              <%= select_tag :gene, options_for_select(load_gene_nav(@genes)), {class: 'form-control'} %>
+              <br />
+              <script type="text/javascript">
+                  $('#gene').change(function() {
+                      if ($(this).val() != '') {
+                          launchModalSpinner('#spinner_target', '#loading-modal');
+                          var url = $(this).val();
+                          var cluster = $('#search_cluster').val();
+                          var annotation = $('#search_annotation').val();
+                          var subsample = $('#search_subsample').val();
+                          $.get( url + '?cluster=' + cluster + '&annotation=' + annotation + '&subsample=' + subsample);
+                      }
+                  });
+              </script>
+            </div>
+          </div>
+          <hr />
+      <% end %>
+      <div class="row">
+        <div class="col-sm-12">
+          <%= form_tag(search_precomputed_results_path(study_name: params[:study_name]), id: 'precomputed-expression', data: {remote: true}) do %>
+              <%= label_tag :expression, 'View curated gene lists as heatmaps' %><br />
+              <%= select_tag :expression, options_for_select(@precomputed, params[:precomputed]), {include_blank: 'Please select a gene list', class: "form-control"} %>
+              <% if action_name !~ /precomputed/ %>
+                  <script type="text/javascript">
+                      $('#expression').change(function() {
+                          if ($(this).val() != '') {
+                              launchModalSpinner('#spinner_target', '#loading-modal');
+                              $('#precomputed-expression').submit();
+                          }
+                      });
+                  </script>
+              <% end %>
+          <% end %>
         </div>
-        <div class="modal-footer">
-          <button class="close" data-dismiss="modal">×</button>
+      </div>
+    </div>
+    <div class="modal fade" id="loading-modal" role="dialog" aria-labelledby="loading-modal" aria-hidden="true">
+      <div class="modal-dialog">
+        <div class="modal-content">
+          <div class="modal-header">
+            <h4 class="text-center">Loading... Please Wait</h4>
+          </div>
+          <div class="modal-body">
+            <div id="spinner_target"></div>
+          </div>
+          <div class="modal-footer">
+            <button class="close" data-dismiss="modal">×</button>
+          </div>
         </div>
       </div>
     </div>
   </div>
+  <% if action_name !~ /heatmap/ %>
+    <div class="panel panel-info" id="search-options-panel">
```

Comment:
I think these might be used for the collapse button, which I still wanted to work without modifying.